### ISSUE TITLE
Patterns in no-restricted-imports not working as expected

### ISSUE THREAD

**Issue by bdougherty** [eslint/eslint#11531]
**Tell us about your environment**

* **ESLint Version:** 5.15.3
* **Node Version:** 11.12.0
* **npm Version:** 6.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"parser": "babel-eslint",
	"rules": {
		"no-restricted-imports": [
			"error",
			{
				"patterns": [
					"@fortawesome/*-svg-icons",
					"!@fortawesome/*-svg-icons/*"
				]
			}
		]
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import { faQrcode } from '@fortawesome/free-solid-svg-icons';
import faCheckCircle from '@fortawesome/free-regular-svg-icons/faCheckCircle';

export default () => {};
```

```bash
eslint src/
```

**What did you expect to happen?**
I am trying to prevent the import of the root package, so I would expect that the first line is an error and the second is not.

**What actually happened? Please include the actual, raw output from ESLint.**
Both lines result in an error:

```
/Users/brad/Downloads/eslint-test/src/index.js
  1:1  error  '@fortawesome/free-solid-svg-icons' import is restricted from being used by a pattern                  no-restricted-imports
  2:1  error  '@fortawesome/free-regular-svg-icons/faCheckCircle' import is restricted from being used by a pattern  no-restricted-imports

✖ 2 problems (2 errors, 0 warnings)
```

If the patterns are reversed, it correctly flags only the second line as an error.

<details>
<summary>Configuration</summary>

```js
{
	"parser": "babel-eslint",
	"rules": {
		"no-restricted-imports": [
			"error",
			{
				"patterns": [
					"@fortawesome/*-svg-icons",
					"!@fortawesome/*-svg-icons/*"
				]
			}
		]
	}
}
```
</details>

```
/Users/brad/Downloads/eslint-test/src/index.js
  2:1  error  '@fortawesome/free-regular-svg-icons/faCheckCircle' import is restricted from being used by a pattern  no-restricted-imports

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**
Sure

**Comment by platinumazure** [comment]
Hi @bdougherty, very sorry this slipped through the cracks.

Could you elaborate on what you mean by "if the patterns are reversed"? Your second configuration is identical to your first configuration, and I don't see another source posting, so I'm a bit confused what change you made that makes the behavior "work".

**Comment by bdougherty** [comment]
Sorry, I guess it was a copy/paste error. I am sure that I meant to reverse the order of the items in the patterns array, but I can't get it to work either way now, with 5.15.3 or 5.16.0.

Ended up not needing this, so I'm fine with it being closed. I can bring it up again if I ever run into it in the future.
